compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -321,6 +352,11 @@ impl<Prov, Extra> Allocation<Prov, Extra> {
 
 /// Byte accessors.
 impl<Prov: Provenance, Extra> Allocation<Prov, Extra> {
+    /// Get the pointer of the [u8] of bytes.
+    pub fn get_bytes_addr(&self) -> Size {
```

Comment:
This is deliberately the *host* address, right? So it should have return type `usize`, not `Size`.

Also, to actually allow C code to use this address, we need to be a bit more careful with aliasing here. The pointer you are exposing here will be invalidated the next time someone mutates this memory, causing UB in Miri itself due to violating the Rust aliasing rules.

Under my proposal where `bytes: Bytes`, I think the `AllocationBytes` trait should have a method `expose_base_addr` with return type `usize`, and this function here simply calls `expose_base_addr`. (It should probably also be called `expose_base_addr`.)

Reply:
Good point -- thanks for the suggestion, that makes sense -- `AllocationBytes` trait is a clean solution.